pandas/tests/indexes/object/test_indexing.py

```diff
@@ -92,6 +93,11 @@ def test_get_indexer_non_unique_nas(self, nulls_fixture):
             tm.assert_numpy_array_equal(indexer, expected_indexer)
             tm.assert_numpy_array_equal(missing, expected_missing)
 
+    @pytest.mark.skipif(
+        parse_version(np.__version__) > parse_version("1.24.0")
+        and "dev" in np.__version__,
+        reason="GH50124",
+    )
```

Comment:
I've marked this one as failing because in the new nightly numpy there's
```python
      1 nat_1 = np.datetime64('NaT', 'Y')
      2 nat_2 = np.datetime64('NaT', 'ps')
----> 3 {nat_1, nat_2}

OverflowError: Integer overflow getting a common metadata divisor for NumPy datetime metadata [Y] and [ps].
```

I don't see why we can't put them both in a set, there shouldn't be any comparision between them going on - reckon this is a bug in numpy? If so, is it OK to skip this test for now?

Reply:
Also, is there a better way to write this condition? `parse_version(np.__version__) > parse_version("1.25.0")` returns `False` unfortunately

Reply:
There's a `pandas.compat.is_numpy_dev`

https://github.com/pandas-dev/pandas/blob/0119bdc57c41a71ae3cdd5fc47f384e504c19208/pandas/compat/__init__.py#L26